### PR TITLE
apply expression and payload transformations  to commands

### PR DESCRIPTION
**Pull request by AlvaroVega**
Issues:  https://github.com/telefonicaid/iotagent-json/issues/634 https://github.com/telefonicaid/iotagent-json/issues/627

related and needs https://github.com/telefonicaid/iotagent-node-lib/pull/1209

These are the possible payloadType:
https://github.com/telefonicaid/iotagent-json/blob/6244896089ef739731888052608c26f4bf051d4c/lib/commandHandler.js#L43-L56

- [x] Update doc
- [x] Content-Type

### CODE REVIEW

**Comment by fgalan** [comment]
> Should a "twin" PR be created in IOTA-UL repository?

Twin PR https://github.com/telefonicaid/iotagent-ul/pull/550

**Comment by AlvaroVega** [comment]
> var b = Buffer.from('FF','HEX')
undefined
> b.length
1
> b[0]

File: docs/usermanual.md [hunk 1]
```diff
@@ -450,6 +452,40 @@ a transformation expression in the IoTAgent). In this cases, when a reverse expr
...
+```
+
+and when command will be executed the command value will be the result of apply value to defined expression. Followig
+the example case the command will be:
+
+```json
+{
+    "set": 0
+}
+```
```
Comment (fgalan) [inline docs/usermanual.md:475]: Should be in UL format? Something like `set|0` (unsure)

File: docs/usermanual.md [hunk 2]
```diff
@@ -450,6 +452,40 @@ a transformation expression in the IoTAgent). In this cases, when a reverse expr
...
+
+Additionally a command could define a `payloadType` in their definnition with the aim to transform payload command with
+the following meanings:
+
+-   **binaryfromstring**: Payload will transformed into a be Buffer after read it form a string.
+-   **binaryfromhex**: Payload will transformed into a be Buffer after read it form a string hex.
+-   **binaryfromjson**: Payload will transformed into a be Buffer after read it form a JSON string.
+-   **json**: This is the default case. Paylaod will be stringify form a JSON.
```
Comment (fgalan) [inline docs/usermanual.md:483]:
```suggestion
-   **json**: This is the default case. Payload will be stringify form a JSON.
```

File: docs/usermanual.md (hunk 1 above)
Comment (fgalan) [reply docs/usermanual.md:475]: You are right... this is the right one: https://github.com/telefonicaid/iotagent-ul/pull/550#discussion_r838661612

(I was reviewing the two PRs in parallel and I put the comment in the wrong place :)

NTC (in this PR)

File: docs/usermanual.md [hunk 3]
```diff
@@ -450,6 +452,40 @@ a transformation expression in the IoTAgent). In this cases, when a reverse expr
...
+and when command will be executed the command value will be the result of apply value to defined expression. Following
+the example case the command will be:
+
+```json
+{
+    "set": 0
+}
+```
+
+Additionally a command could define a `payloadType` in their definnition with the aim to transform payload command with
```
Comment (fgalan) [inline docs/usermanual.md:477]:
```suggestion
Additionally a command could define a `payloadType` in their definition with the aim to transform payload command with
```

**Review by fgalan — APPROVED** [review]
LGTM (but waiting to merge https://github.com/telefonicaid/iotagent-node-lib/pull/1209 before merging this one)

**Pull request merged**